pkg/operator/controller/canary/controller.go

```diff
@@ -189,6 +230,10 @@ func (r *reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
 		r.mu.Lock()
 		r.enableCanaryRouteRotation = v
 		r.mu.Unlock()
+	} else {
+		r.mu.Lock()
+		r.enableCanaryRouteRotation = false
+		r.mu.Unlock()
 	}
```

Comment:
Both the if and else branch could be consolidated to: 

```suggestion
	val, ok := ic.Annotations[CanaryRouteRotationAnnotation]
	v, _ := strconv.ParseBool(val)
	r.mu.Lock()
	r.enableCanaryRouteRotation = ok && v
	r.mu.Unlock()
```

Or using defer if you prefer.